### PR TITLE
Mouse View

### LINKED ISSUE #28889: Please restore Mouse View functionality to the sidebar.
Please restore Mouse View functionality to the sidebar.

Prior to the new customisable sidebar layout Mouse View used to appear whenever the mouse was moved from the sidebar into the main window, temporarily covering the compass and part of the message log.
It automatically disappeared upon movement of the player character, or the mouse returning to the sidebar.

If the old method of displaying it isn't possible under the new layout please could it be added as an optional panel in the sidebar.
(It could default to off, if you are concerned about preventing clutter in the default view)

I'm aware we can still use mouse look via opening the Look Around pop-up menu with ; or x, however this adds unnecessary repeated keystrokes, including having to press ESC to close the pop-up, each time we start and finish looking around.

**Versions and configuration(please complete the following information):**
 - OS: Windows 7
 - Game Version: 0.D-841-g462722b
 - Graphics version: Tiles

### PR DESCRIPTION
**Pull request by RoyBerube**
Mouse view appears when hovering mouse on terrain. Cancels on any other user input.

#### Summary
SUMMARY: Interface "Mouse view shows tile contents when hovering mouse."

#### Purpose of change
Fixes #28889 

#### Describe the solution
Adds back feature missing since 0.D stable. Leverages existing code to display terrain info in a window that resizes according to the amount of information.

#### Testing
Tested in Windows only.

#### Additional context
User has no way to reposition the window. It is currently set to start at the top of the info panel.

### CODE REVIEW

File: src/game.cpp [hunk 1]
```diff
@@ -2165,16 +2165,24 @@ int game::inventory_item_menu( int pos, int iStartX, int iWidth,
...
+    int width = sidebar_right ? mgr.get_width_right() : mgr.get_width_left();
     do {
         action = ctxt.handle_input();
         if( action == "MOUSE_MOVE" ) {
             const cata::optional<tripoint> mouse_pos = ctxt.get_coordinates( w_terrain );
             if( mouse_pos && ( !liveview_pos || *mouse_pos != *liveview_pos ) ) {
                 liveview_pos = mouse_pos;
                 liveview.show( *liveview_pos );
+                draw_panels( true );
+                catacurses::window &w = catacurses::newwin( TERMY / 2, width,
```
Comment (Vollinger) [inline src/game.cpp]:
```suggestion
                const catacurses::window &w = catacurses::newwin( TERMY / 2, width,
```
GCC complained about `invalid initialization of non-const reference of type 'catacurses::window&' from an rvalue of type 'catacurses::window'`

Comment (RoyBerube) [reply src/game.cpp]: Changed. Tested to make sure it still works the same.

File: src/game.cpp [hunk 2]
```diff
@@ -2165,16 +2165,24 @@ int game::inventory_item_menu( int pos, int iStartX, int iWidth,
...
         action = ctxt.handle_input();
         if( action == "MOUSE_MOVE" ) {
             const cata::optional<tripoint> mouse_pos = ctxt.get_coordinates( w_terrain );
             if( mouse_pos && ( !liveview_pos || *mouse_pos != *liveview_pos ) ) {
                 liveview_pos = mouse_pos;
                 liveview.show( *liveview_pos );
+                draw_panels( true );
+                const catacurses::window &w = catacurses::newwin( TERMY / 2, width,
+                                        point( sidebar_right ? TERMX - width : 0, 0 ) );
+                g->liveview.draw( w, TERMY / 2 );
```
Comment (Qrox) [inline src/game.cpp]:
```suggestion
                liveview.draw( w, TERMY / 2 );
```
This is already a method of `game`.

**Comment by MetalSpork** [comment]
@RoyBerube Don't forget to claim the bounty. https://www.bountysource.com/issues/71726642-25-please-restore-mouse-view-functionality-to-the-sidebar

**Comment by ymber** [comment]
Looks like this introduced #35324.

**Pull request merged**